### PR TITLE
Live Photo

### LINKED ISSUE #12124: Implement live photo
See https://github.com/nextcloud/server/issues/40242#issuecomment-1737934737

### PR DESCRIPTION
**Pull request by alperozturk96**
Resolves #12124 

Live Photo Indicator List View:
list_

Live Photo Indicator Grid View:
grid_

Live Photo Indicator List View SFSymbol Alternative:

Live Photo Indicator Grid View SFSymbol Alternative:

@nextcloud/designers How does the live photo indicator look? I think the position and color can be better, but nothing come in my mind in this regard.

- [x] Tests written, or not not needed

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
@alperozturk96 the icon is: https://developer.apple.com/live-photos/ So I would just use that, no?

See https://thenounproject.com/browse/icons/term/live-photo/ for svg resources, maybe @jancborchardt or @nimishavijay can pick one? :)

**Comment by alperozturk96** [comment]
@AndyScherzinger I agree that icon more suitable. @tobiasKaminsky [suggested](https://github.com/nextcloud/server/issues/40242#issuecomment-1770405160) that icon. If it's okay for all of us I can change.

**Comment by jancborchardt** [comment]
This icon seems to be the closest to Apple’s live photo icon: https://thenounproject.com/icon/play-live-photo-710577/

**Comment by alperozturk96** [comment]
Icons updated @AndyScherzinger @jancborchardt

File: app/src/main/res/layout/list_item.xml [hunk 1]
```diff
@@ -241,19 +253,18 @@
...
             android:contentDescription="@string/checkbox"
             android:focusable="false"
-            android:minHeight="@dimen/min_list_item_size"
             android:paddingStart="@dimen/standard_half_padding"
             android:paddingEnd="@dimen/alternate_padding"
             android:src="@drawable/ic_checkbox_blank_outline" />
 
         <ImageView
             android:id="@+id/overflow_menu"
-            android:layout_width="48dp"
```
[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply app/src/main/res/layout/list_item.xml:256]: There is a dimens value for that also 👍

**Comment by AndyScherzinger** [comment]
@jancborchardt i.e. amazon is showing it, which I think makes sense, since we are not the Apple Image viewer but a general EFSS in this case, so _if_ we then hide the mov file in these file lists than we might want to indicate a live photo as such, no?

**Comment by AndyScherzinger** [comment]
For the overlay in the grid view this definitely needs a background container to "punch out" the icon, else it is extremely hard to spot. So I would suggest just using the background color (surface), semi-transparent might look strange, on the other hand the share icon has the same issue too.

@jancborchardt

**Comment by jancborchardt** [comment]
Ok, then if we show it I would say
- Position: bottom left of the preview, with some padding to the sides
- Style-wise do it like Apple, semi-transparent grey backdrop so the icon has contrast

**Comment by AndyScherzinger** [comment]
> Style-wise do it like Apple, semi-transparent grey backdrop so the icon has contrast

@jancborchardt can maybe @marcoambrosini check how Apple displays it on dark mode (since I only have Android)

**Comment by marcoambrosini** [comment]
@AndyScherzinger is this what you need? 

Screenshot 2023-11-07 at 10 46 49

**Comment by AndyScherzinger** [comment]
> @AndyScherzinger is this what you need?

perfect, yes, thanks @marcoambrosini 👍

File: app/src/main/java/com/owncloud/android/db/ProviderMeta.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
  */
 public class ProviderMeta {
     public static final String DB_NAME = "filelist";
-    public static final int DB_VERSION = 73;
+    public static final int DB_VERSION = 75;
```
Comment (AndyScherzinger) [inline app/src/main/java/com/owncloud/android/db/ProviderMeta.java:38]: DB version bump by 2? Also where is the new json schema for DB?

**Comment by alperozturk96** [comment]
@AndyScherzinger @tobiasKaminsky @jancborchardt 

https://github.com/nextcloud/android/assets/67455295/8ccacefc-0055-4e66-b4af-8d0af5347ce6


I added demo so you can see current behaviour of application. From backend we are getting 2 file one for video and other one is image. In app I am adding video file inside the photo file and hide video file from list because it's actually one file for user but for us it's two.

In current implementation video of live photo automatically is playing.

I suggest we shouldn't automatically play video of live photo. When user opened the file it should just show the image and if user long press to the image then we can play the video of live photo. By automatically playing the live photo's video, we essentially diminish the photo aspect of the live photo, rendering the indicators superfluous. We already have indicator in the list and top left side of image, so user will know about it. Long-pressing is a familiar gesture for both Android and iOS users.

We can also introduce a new action, like a "play" option accessible from the top-right three dots menu, to make it easy for users to play the video of live photo if they choose to do so.

**Comment by AndyScherzinger** [comment]
Looks good so far. 👍

Some comments

* date in file listing is misaligned for live photo
* live photo view shows 2 3-dot menus
* see discussion in live photo chat, server team plans to add a `hidden` dav property which is requestable by clients reflecting if a file should be hidden, so the server would calculate this based on the live photo meta data

**Comment by tobiasKaminsky** [comment]
> I suggest we shouldn't automatically play video of live photo.

With having this "live" button on top left, I think it is also better to not play it immediatly.
@jancborchardt ?

File: settings.gradle [hunk 3]
```diff
@@ -8,3 +8,11 @@ include ':appscan'
 //        substitute module('com.github.nextcloud.android-common:ui') using project(':ui')
 //    }
 //}
+
+
+// It will then be disabled as a comment.
+includeBuild('/Users/alperozturk/Desktop/nextcloud/nextcloud_android_library') {
```
Comment (AndyScherzinger) [inline settings.gradle:14]: this should not be commited

**Comment by alperozturk96** [comment]
@tobiasKaminsky This PR waiting this [PR](https://github.com/nextcloud/android-library/pull/1237) fyi

**Comment by tobiasKaminsky** [comment]
```
> Task :app:compileGplayDebugAndroidTestKotlin
e: file:///Users/runner/work/android/android/app/src/androidTest/java/com/owncloud/android/ui/fragment/UnifiedSearchFragmentIT.kt:42:53 No value passed for parameter 'listOfHiddenFiles'
e: file:///Users/runner/work/android/android/app/src/androidTest/java/com/owncloud/android/ui/fragment/UnifiedSearchFragmentIT.kt:75:53 No value passed for parameter 'listOfHiddenFiles'
```

Please always check for your PR why a test fails.
I will fix it in this case.

**Review by tobiasKaminsky — APPROVED** [review]

**Comment by JonasMayerDev** [comment]
Hi, I noticed some bugs.
1. Live icon seems odd to me when live photo is opened (is there supposed to be a text?)
2. Clicking on live icon multiple times when opened crashes app (and leads to unexpected behavior) -> Just not listen to clicks when video is playing?

https://github.com/nextcloud/android/assets/43114340/8ebc5d97-e216-49f8-9ba2-d3bdc4ab13b7


3. Options Button in opened live photo crashes the app for me. 

https://github.com/nextcloud/android/assets/43114340/ea8f6080-7e31-4de7-9693-44e604efa7ea

**Comment by alperozturk96** [comment]
> Hi, I noticed some bugs.
> 
> 1. Live icon seems odd to me when live photo is opened (is there supposed to be a text?)
> 2. Clicking on live icon multiple times when opened crashes app (and leads to unexpected behavior) -> Just not listen to clicks when video is playing?
> 
>  Record_2023-12-06-12-04-08.mp4 
> 3. Options Button in opened live photo crashes the app for me.
> 
>  Record_2023-12-06-12-04-31.mp4

Could you share crash report?

**Comment by JonasMayerDev** [comment]
For 2.:
`
2023-12-06 13:11:22.543 14499-14499 AndroidRuntime          com.nextcloud.android.qa             E  FATAL EXCEPTION: main
                                                                                                    Process: com.nextcloud.android.qa, PID: 14499
                                                                                                    java.lang.NullPointerException: Attempt to read from field 'com.google.android.exoplayer2.ui.StyledPlayerView com.owncloud.android.databinding.FragmentPreviewMediaBinding.exoplayerView' on a null object reference in method 'void com.owncloud.android.ui.preview.PreviewMediaFragment.lambda$onStart$1()'
                                                                                                    	at com.owncloud.android.ui.preview.PreviewMediaFragment.lambda$onStart$1(PreviewMediaFragment.java:370)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewMediaFragment.$r8$lambda$AkxMJlf2KgePzqOIh5h57BVdLW8(Unknown Source:0)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewMediaFragment$$ExternalSyntheticLambda0.run(Unknown Source:2)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:938)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:233)
                                                                                                    	at android.os.Looper.loop(Looper.java:344)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8248)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:589)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1071)
`

For 3.:
`
2023-12-06 13:14:26.365 12650-12650 AndroidRuntime          com.nextcloud.android.qa             E  	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)
                                                                                                    	at android.app.Activity.onMenuItemSelected(Activity.java:4404)
                                                                                                    	at androidx.activity.ComponentActivity.onMenuItemSelected(ComponentActivity.java:528)
                                                                                                    	at androidx.fragment.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:264)
                                                                                                    	at androidx.appcompat.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:269)
                                                                                                    	at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:110)
                                                                                                    	at androidx.appcompat.app.AppCompatDelegateImpl.onMenuItemSelected(AppCompatDelegateImpl.java:1259)
                                                                                                    	at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:836)
                                                                                                    	at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:159)
                                                                                                    	at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:987)
                                                                                                    	at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:977)
                                                                                                    	at androidx.appcompat.widget.ActionMenuView.invokeItem(ActionMenuView.java:625)
                                                                                                    	at androidx.appcompat.view.menu.ActionMenuItemView.onClick(ActionMenuItemView.java:156)
                                                                                                    	at android.view.View.performClick(View.java:7498)
                                                                                                    	at android.view.View.performClickInternal(View.java:7471)
                                                                                                    	at android.view.View.access$3700(View.java:843)
                                                                                                    	at android.view.View$PerformClick.run(View.java:29098)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:938)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:233)
                                                                                                    	at android.os.Looper.loop(Looper.java:344)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8248)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
2023-12-06 13:14:26.365 12650-12650 AndroidRuntime          com.nextcloud.android.qa             E  	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:589)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1071)`

I get "com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)" a hundred time befor app actually crashes for some reason (or logger bug?)

**Comment by alperozturk96** [comment]
> For 2.: `2023-12-06 13:11:22.543 14499-14499 AndroidRuntime com.nextcloud.android.qa E FATAL EXCEPTION: main Process: com.nextcloud.android.qa, PID: 14499 java.lang.NullPointerException: Attempt to read from field 'com.google.android.exoplayer2.ui.StyledPlayerView com.owncloud.android.databinding.FragmentPreviewMediaBinding.exoplayerView' on a null object reference in method 'void com.owncloud.android.ui.preview.PreviewMediaFragment.lambda$onStart$1()' at com.owncloud.android.ui.preview.PreviewMediaFragment.lambda$onStart$1(PreviewMediaFragment.java:370) at com.owncloud.android.ui.preview.PreviewMediaFragment.$r8$lambda$AkxMJlf2KgePzqOIh5h57BVdLW8(Unknown Source:0) at com.owncloud.android.ui.preview.PreviewMediaFragment$$ExternalSyntheticLambda0.run(Unknown Source:2) at android.os.Handler.handleCallback(Handler.java:938) at android.os.Handler.dispatchMessage(Handler.java:99) at android.os.Looper.loopOnce(Looper.java:233) at android.os.Looper.loop(Looper.java:344) at android.app.ActivityThread.main(ActivityThread.java:8248) at java.lang.reflect.Method.invoke(Native Method) at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:589) at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1071)`
> 
> For 3.: ` 2023-12-06 13:14:26.365 12650-12650 AndroidRuntime com.nextcloud.android.qa E at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226) at android.app.Activity.onMenuItemSelected(Activity.java:4404) at androidx.activity.ComponentActivity.onMenuItemSelected(ComponentActivity.java:528) at androidx.fragment.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:264) at androidx.appcompat.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:269) at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:110) at androidx.appcompat.app.AppCompatDelegateImpl.onMenuItemSelected(AppCompatDelegateImpl.java:1259) at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:836) at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:159) at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:987) at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:977) at androidx.appcompat.widget.ActionMenuView.invokeItem(ActionMenuView.java:625) at androidx.appcompat.view.menu.ActionMenuItemView.onClick(ActionMenuItemView.java:156) at android.view.View.performClick(View.java:7498) at android.view.View.performClickInternal(View.java:7471) at android.view.View.access$3700(View.java:843) at android.view.View$PerformClick.run(View.java:29098) at android.os.Handler.handleCallback(Handler.java:938) at android.os.Handler.dispatchMessage(Handler.java:99) at android.os.Looper.loopOnce(Looper.java:233) at android.os.Looper.loop(Looper.java:344) at android.app.ActivityThread.main(ActivityThread.java:8248) at java.lang.reflect.Method.invoke(Native Method) 2023-12-06 13:14:26.365 12650-12650 AndroidRuntime com.nextcloud.android.qa E at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:589) at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1071)`
> 
> I get "com.owncloud.android.ui.preview.PreviewImageActivity.onOptionsItemSelected(PreviewImageActivity.java:226)" a hundred time befor app actually crashes for some reason (or logger bug?)

This bug fixed in this [PR](https://github.com/nextcloud/android/pull/12226) already

**Pull request merged**